pkg/operator/workload/workload_openshiftapiserver_v311_00_sync.go

```diff
@@ -105,16 +106,33 @@ func NewOpenShiftAPIServerWorkload(
 // PreconditionFulfilled is a function that indicates whether all prerequisites are met and we can Sync.
 func (c *OpenShiftAPIServerWorkload) PreconditionFulfilled() (bool, error) {
 	ctx := context.TODO() // needs support in library-go
-	originalOperatorConfig, err := c.operatorConfigClient.OpenShiftAPIServers().Get(ctx, "cluster", metav1.GetOptions{})
+	operatorConfig, err := c.operatorConfigClient.OpenShiftAPIServers().Get(ctx, "cluster", metav1.GetOptions{})
 	if err != nil {
 		return false, err
 	}
-	operatorConfig := originalOperatorConfig.DeepCopy()
+	return c.preconditionFulfilledInternal(operatorConfig)
+}
 
+func (c *OpenShiftAPIServerWorkload) preconditionFulfilledInternal(operator *operatorv1.OpenShiftAPIServer) (bool, error) {
 	// block until config is obvserved
-	if len(operatorConfig.Spec.ObservedConfig.Raw) == 0 {
+	if len(operator.Spec.ObservedConfig.Raw) == 0 {
 		klog.Info("Waiting for observed configuration to be available")
-		return false, nil
+		return false, errors.New("waiting for observed configuration to be available")
+	}
+
+	// specifying etcd servers list is mandatory, without it the pods will be crashlooping, so wait for it.
+	// TODO: once OAS moves to using args directly (not via the config)
+	//       we could have a helper function for checking mandatory args for oauth-apiserver and openshift-apiserver
+	//       alternatively we could change the generic workload ctrl to accept a list of common preconditions
+
+	var typedObservedConfig openshiftcontrolplanev1.OpenShiftAPIServerConfig
```

Comment:
it is what we have in the config, if we don't like it we could unmarshal it to a map and get only what we need or read it from a serialized object